rwkv/measure_pexplexity.py

```diff
@@ -17,20 +16,30 @@ def parse_args():
     parser.add_argument('model_path', help='Path to model checkpoint file', type=str)
     parser.add_argument('text_path', help='Path to text file in UTF-8 encoding', type=str)
     parser.add_argument('ignore_first_n_tokens', help='How many tokens should be skipped before loss is measured', type=int)
-    parser.add_argument('token_limit', help='How many tokens to process; set to -1 to process all text', nargs='?', type=int, default=-1)
+    parser.add_argument('token_limit', nargs='?', help='How many tokens to process; set to -1 to process all text', type=int, default=-1)
+    parser.add_argument('tokenizer', help='Which tokenizer to use', nargs='?', type=str, default="20b")
     return parser.parse_args()
 
 args = parse_args()
 
-# ---
-
-print('Loading 20B tokenizer')
-tokenizer_path: pathlib.Path = pathlib.Path(os.path.abspath(__file__)).parent / '20B_tokenizer.json'
-tokenizer: tokenizers.Tokenizer = tokenizers.Tokenizer.from_file(str(tokenizer_path))
-
 print('Loading text')
 text: str = open(args.text_path, encoding='utf-8').read()
-tokens: List[int] = tokenizer.encode(text).ids
+
+if args.tokenizer == "world":
+    print('Loading world tokenizer')
+    import rwkv_tokenizer
+    tokenizer = rwkv_tokenizer.TRIE
+    tokens = tokenizer.encode(text)
+elif args.tokenizer == "20b":
+    print('Loading 20B tokenizer')
+    import tokenizers
+    tokenizer_path: pathlib.Path = pathlib.Path(os.path.abspath(__file__)).parent / '20B_tokenizer.json'
+    tokenizer: tokenizers.Tokenizer = tokenizers.Tokenizer.from_file(str(tokenizer_path))
+    tokens: List[int] = tokenizer.encode(text).ids
+else:
+    print(f"Unknown tokenizer: {args.tokenizer}")
+    quit()
```

Comment:
This `if` statement was repeated three times, I propose moving it into a separate utility file and share between scripts.

Reply:
Should this be a separate file or should it be moved into rwkv_tokenizer.py?

Reply:
Hm, moving to existing file `rwkv_tokenizer.py` sounds okay to me.